### ISSUE TITLE
Option to open the resource again

### ISSUE THREAD

**Issue by tinchoz49** [mafintosh/nanoresource#3]
Currently, once the resource is close it cannot be open it again, we have to create a new resource. I understand that it follows the same principle of interfaces like fs but maybe it could be useful to have an option to allow open it again. 

A good case is the interface of levelup that allows this operation.

**Comment by mafintosh** [comment]
Yep. If it's simple, ie, doesn't complicate or slow down the state machine, i'm for it

**Comment by tinchoz49** [comment]
What do you think about this PR https://github.com/mafintosh/nanoresource/pull/4?

If you prefer to keep the original code I can move the reopen feature into a new branch.

**Comment by tinchoz49** [comment]
related: https://github.com/mafintosh/nanoresource/pull/5
